BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/DistanceUtil.java

```diff
@@ -21,4 +27,25 @@ public static int getMaximumEvenIncrement(double distance) {
         return (int) Math.pow(10, Math.floor(Math.log10(distance)));
     }
 
+    public static Location walkingSpeedFilter(Location oldLocation, Location newLocation) {
+        return speedFilter(oldLocation, newLocation, HUMAN_WALKING_SPEED);
+    }
+
+    /**
+     * Define a maximum movement speed to restrain a new location being further away than the
+     * distance possible in that time window.
+     */
+    public static Location speedFilter(Location oldLocation, Location newLocation, double maximumSpeed) {
```

Comment:
Using this may result in the application being greatly wrong as we will set the location back continuously if we walk faster as the defined `maximumSpeed`.

Reply:
True. It could be helpful to reduce multipath signal error and stabilize the calculated location.

Reply:
But still, how will we handle these continuous error? Setting back the `location` and using this adjusted (incorrect) `location` the next time we have an location update it can delay the opening of `gate`'s for multiple location update intervals (therefore it may delay the trigger for a few seconds).